spec/std/string_spec.cr

```diff
@@ -1444,6 +1444,7 @@ describe "String" do
     "a".dump.should eq %("a")
     "\\".dump.should eq %("\\\\")
     "\"".dump.should eq %("\\\"")
+    # TODO: "\a".dump.should eq %("\\a")
```

Comment:
Maybe add a note `uncomment after 0.24.2`?

Reply:
I think this is enough. Right after 0.24.2 gets released, I will finish this with another PR and there I will just look here to see what I need to do.